apiary.apib

```diff
@@ -4455,3 +4455,156 @@ Gets stats for a given Run Id. Provides the count of imported/exported tables an
                     }
                 }
             }
+# Group Triggers
+
+###Create [POST /v2/storage/triggers/]
+Created a trigger listeng on set of tables
+
++ Attributes
+    + runWithTokenId (required, number) - Id of token used for running configured component
+    + componentType: `orchestration` (required, enum[string]) - For now we support only 'orchestration'
+    + configurationId (required, number) - Id of component configuration
+    + coolDownPeriod (required, number) - Minimal cool down period before firing action again in minutes (min is 5 minutes)
+    + tabledIds (required, object) - Array of tables ids
+        + tableId (string) - Id of table (like "in.c-test.watched")
++ Request
+
+    + Headers
+
+            X-StorageApi-Token: your_token
+
+    + Body
+
+            runWithTokenId=123&componentType=orchestration&configurationId=123&coolDownPeriod=20&tableIds[0]=in.c-test.watched-1&tableIds[1]=in.c-prod.watched-5
+
++ Response 201 (application/json)
+
+        {
+            "id": 3,
+            "runWithTokenId": 123,
+            "projectId": 234,
+            "componentType": "orchestration",
+            "configurationId": 123,
+            "lastRun": null,
+            "coolDownPeriod": 20,
+            "creatorName": "Name Surname",
+            "creatorId": 1,
+            "createdByTokenName": "dev@keboola.com",
+            "createdByTokenId": 4,
+            "tables": [
```

Comment:
Ty interní bucketId na venek nepožíváme. Pracoval bych tady se standardními idčky tabulek https://keboola.docs.apiary.io/#reference/tables/manage-tables/table-detail

Takže by to mohlo být:
```
 "tables": [
  "in.c-main.watched",
  "out.c-neco.watched-5"
]
```

Případně rozšiřitelnější formát, ale vzhledem k tomu že se to tam zadává taky jako pole tak je to možná zbytečný.
```
 "tables": [
  {
   "tableId":  "in.c-main.watched"
  },
 {
  "tableId": "out.c-neco.watched-5"
 }
]
```

Reply:
taky jsem si to říkal. Tohle jsem udělal, protože primární klíč (bucketId + name) a to id se ve vazbě vůbec nevyskytuje. Tohle je výpis z vazební tabulky trigger <> table, kde je jenom buckerId, name a triggerId

Reply:
Udělal jsem solo metodu s přímým query jenom pro tu API response.